### ISSUE TITLE
MOM_W Strategy Implementation

### ISSUE THREAD

**Issue by PaiCY-T** [PaiCY-T/ML4T#47]
---
epic: ML4T
task_number: 004
title: MOM_W Strategy Implementation
description: Single momentum strategy with LightGBM ML enhancement using FinLab MLEngine
status: pending
created: 2025-09-22T00:02:15Z
updated: 2025-09-22T00:02:15Z
assignee: null
depends_on: [001, 002, 003]
parallel: true
size: L
estimated_hours: 80
---

# Task 004: MOM_W Strategy Implementation

## Overview

Implement a comprehensive weekly momentum strategy with machine learning enhancement using FinLab's MLEngine. This strategy will serve as one of the core alpha-generating components in the multi-strategy portfolio system.

## Objectives

- Build weekly momentum strategy using technical indicators (RSI, SMA ratios, volatility)
- Integrate LightGBM model for signal enhancement via FinLab MLEngine
- Implement proper time-series cross-validation for ML model training
- Generate robust weekly rebalancing signals with risk controls

### In Scope
- Weekly momentum factor calculation and ranking
- LightGBM model training and prediction pipeline
- Time-series cross-validation framework
- Signal generation and position sizing logic
- Integration with FinLab data infrastructure
- Basic risk controls (position limits, sector exposure)

### Out of Scope
- Multi-strategy portfolio allocation (covered in other tasks)
- Complex regime detection models
- Alternative ML models beyond LightGBM
- Real-time execution infrastructure

### Data Requirements
- Weekly OHLCV data for TSE/OTC universe
- Technical indicators: RSI(14), SMA(20), 20-day volatility
- Fundamental ratios: P/B, ROE for additional features
- Liquidity filters: average daily volume, market cap

### ML Model Specifications
- Model: LightGBM with FinLab MLEngine integration
- Target: 21-day forward excess return over market mean
- Features: momentum factors, technical indicators, fundamental ratios
- Cross-validation: Time-series split with 3-year training, 1-year validation
- IC threshold: Minimum 0.02 Information Coefficient for factor inclusion

### Strategy Logic
- Universe: TSE/OTC stocks with liquidity filters
- Rebalancing: Weekly (every Monday)
- Entry: Top 30 stocks by composite ML score
- Exit: Drop from top 30 or RSI < 30 signal
- Position sizing: Equal weight with 10% individual position limit

### Phase 1: Core Infrastructure (Week 1-2)
- Set up weekly data pipeline using FinLab data.get()
- Implement technical indicator calculations
- Create base strategy class structure

### Phase 2: Feature Engineering (Week 3-4)
- Build comprehensive momentum factor library
- Implement feature selection and IC analysis
- Create data preprocessing pipeline for ML model

### Phase 3: ML Model Development (Week 5-7)
- Integrate FinLab MLEngine for LightGBM training
- Implement time-series cross-validation framework
- Develop model evaluation and selection pipeline
- Optimize hyperparameters for momentum prediction

### Phase 4: Strategy Integration (Week 8-10)
- Combine traditional momentum ranking with ML scores
- Implement position sizing and risk controls
- Create signal generation and validation framework
- Build performance tracking and attribution

### Code Components
- `src/strategies/mom_w_strategy.py` - Main strategy implementation
- `src/ml/lightgbm_engine.py` - ML model training and prediction
- `src/factors/momentum_factors.py` - Momentum factor library
- `src/utils/time_series_cv.py` - Cross-validation framework

### Configuration
- `config/strategies/mom_w.yml` - Strategy parameters and settings
- `config/ml/lightgbm_config.yml` - ML model configuration

### Documentation
- Strategy methodology and factor definitions
- ML model architecture and training procedure
- Performance attribution framework
- Risk control implementation

### Performance Targets
- Information Coefficient (IC) > 0.05 for momentum factors
- ML model validation accuracy with IC > 0.03
- Strategy Sharpe ratio > 1.5 in backtesting (indicative)
- Maximum drawdown < 20% for standalone strategy

### Technical Quality
- 90%+ test coverage for core strategy logic
- Sub-second signal generation for 2000+ stock universe
- Robust handling of missing data and corporate actions
- Integration with existing data infrastructure

### Operational Requirements
- Automated weekly signal generation
- Comprehensive logging and error handling
- Configuration-driven parameter management
- Clear performance attribution and reporting

## Dependencies

- Task 001: Core data infrastructure and FinLab integration
- Task 002: Base strategy framework and interfaces
- Task 003: ML infrastructure and evaluation framework

### Technical Risks
- **Data quality issues**: Implement comprehensive data validation
- **Model overfitting**: Use robust time-series CV and regularization
- **Performance degradation**: Regular model retraining and monitoring

### Market Risks
- **Momentum factor decay**: Monitor IC and implement factor rotation
- **Regime changes**: Build adaptive model selection framework
- **Liquidity constraints**: Implement dynamic universe filters

## Notes

- Strategy designed for Taiwan market with local cost structure
- Focus on simplicity and robustness over complexity
- Emphasize explainable factors for risk management
- Plan for future integration with multi-strategy framework
